results/cmd/watcher/main.go

```diff
@@ -108,12 +113,58 @@ func (r *reconciler) Reconcile(ctx context.Context, key string) error {
 		r.logger.Errorf("Error converting to proto: %v", err)
 		return err
 	}
-	if _, err := r.client.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{
-		TaskRun: p,
-	}); err != nil {
-		r.logger.Error("Error updating TaskRun %s: %v", name, err)
-		return err
-	}
 
+	// Create a TaskRun if it does not exist in results server, update existing one otherwise.
+	annotation := p.GetMetadata().GetAnnotations()
+	if val, ok := annotation["results_id"]; ok {
+		if _, err := r.client.UpdateTaskRunResult(ctx, &pb.UpdateTaskRunRequest{
+			TaskRun:   p,
+			ResultsId: val,
+		}); err != nil {
+			r.logger.Error("Error updating TaskRun %s: %v", name, err)
+			return err
+		}
+		r.logger.Infof("update id ", val)
+	} else {
+		res, err := r.client.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{
+			TaskRun: p,
+		})
+		if err != nil {
+			r.logger.Error("Error creating TaskRun %s: %v", name, err)
+			return err
+		}
+
+		// Connect to gke cluster and add results_id to annotations field of TaskRun
+		loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+		configOverrides := &clientcmd.ConfigOverrides{}
+		kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+		config, err := kubeConfig.ClientConfig()
+		if err != nil {
+			r.logger.Error("Error creating client config for TaskRun %s: %v", name, err)
+		}
+		client, err := taskinterface.NewForConfig(config)
+		if err != nil {
+			r.logger.Error("Error creating client for TaskRun %s: %v", name, err)
+		}
```

Comment:
These are things that can be set up once for the reconciler, instead of on each reconcile event. We should do this on reconciler creation instead.

Reply:
done